build.go

```diff
@@ -36,21 +38,33 @@ func (r *Runner) Command(command string, args ...string) error {
 	return nil
 }
 
-func (r *Runner) UpdateYum() error {
-	return r.Command("yum", "update -y")
-}
+func (r *Runner) InstallPackage(packages []string) error {
+
+	if !r.updated {
+		r.Command("yum", "update -y")
+		r.updated = true
+	}
 
-func (r *Runner) InstallPackages(packages []string) error {
 	for _, p := range packages {
-		err := r.Command("yum", "install -y "+p)
-		if err != nil {
-			return err
+		if p != "update" {
+			err := r.Command("yum", "install -y "+p)
+			if err != nil {
+				return err
+			}
 		}
 	}
 	return nil
 }
 
 func (r *Runner) InstallGems(packages []string) error {
+
+	if !r.installed {
+		log.Println("Need to install Ruby")
+		r.InstallPackage([]string{"ruby"})
```

Comment:
Do we also need to install the `gem` package here, or do we get it automatically when we install `ruby`?

Reply:
I was thinking to include a test to check if ruby is installed first prior to installing `bundler` and `ruby`. The gems packages will install automatically with ruby. 

Hmm, thinking about it, do we really need to check it, as you previously stated, 'r.Command()' will let us know when it doesn't like something.

Reply:
Well, we were already thinking it might be useful to have an `IsPackageInstalled()` method, weren't we, so if we had that, then we could use it for free here!